### PR TITLE
feat/PRSD-896 update number of households validation

### PR DESCRIPTION
**Pull request by Kiran-R-K**
changed the validation on number of household for the property registration to be a positive-integer validator rather than a positive-or-zero-integer validator

Note: the message key does not need to be updated

image

### CODE REVIEW

**Review by isobel-softwire — APPROVED** [review]
LGTM!

**Comment by Travis-Softwire** [comment]
Worth adding a test (or adding to an existing test) to check for the `0` edge case?

File: src/main/kotlin/uk/gov/communities/prsdb/webapp/validation/PositiveIntegerValidator.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import org.hibernate.validator.internal.constraintvalidators.bv.number.sign.PositiveValidatorForInteger
+
+class PositiveIntegerValidator : PropertyConstraintValidator {
+    override fun isValid(value: Any?): Boolean {
+        try {
+            val integerValue = value.toString().toInt()
+            return PositiveValidatorForInteger().isValid(integerValue, null)
+        } catch (e: NumberFormatException) {
+            return false
+        }
```
Comment (AEPR) [inline src/main/kotlin/uk/gov/communities/prsdb/webapp/validation/PositiveIntegerValidator.kt:12]: I think it would be better to avoid using a `try-catch` block to map to `true` or `false` if possible. I'd use `toIntOrNull` and check if it's returned null instead.

**Review by AEPR — CHANGES_REQUESTED** [review]

File: src/main/kotlin/uk/gov/communities/prsdb/webapp/validation/PositiveIntegerValidator.kt (hunk 1 above)
Comment (isobel-softwire) [reply src/main/kotlin/uk/gov/communities/prsdb/webapp/validation/PositiveIntegerValidator.kt:12]: That sounds good to me. I think the same change should be made in the other numerical validators (`DateValidator` and `PositiveOrZeroIntegerValidator`) as well

**Comment by AEPR** [comment]
I accidentally pushed a change to this branch that I didn't intend to - so I force-pushed to revert it back to the correct commit.
